core/execution/common/liquidity_provision.go

```diff
@@ -0,0 +1,671 @@
+		return ErrCommitmentSubmissionNotAllowed
+	}
+
+	if err := m.ensureMinCommitmentAmount(sub.CommitmentAmount); err != nil {
+		return err
+	}
+
+	submittedImmediately, err := m.liquidityEngine.SubmitLiquidityProvision(ctx, sub, party, m.idGen)
+	if err != nil {
+		return err
+	}
+
+	if err := m.makePerPartyAccountsAndTransfers(ctx, party, sub.CommitmentAmount); err != nil {
+		if newErr := m.liquidityEngine.RejectLiquidityProvision(ctx, party); newErr != nil {
+			m.log.Debug("unable to submit cancel liquidity provision submission",
+				logging.String("party", party),
+				logging.String("id", deterministicID),
+				logging.Error(newErr))
+
+			err = fmt.Errorf("%v, %w", err, newErr)
+		}
+
+		return err
+	}
+
+	if submittedImmediately {
+		// now we can setup our party stake to calculate equities
+		m.equityShares.SetPartyStake(party, sub.CommitmentAmount.Clone())
+		// force update of shares so they are updated for all
+		_ = m.equityShares.AllShares()
+	}
+
+	return nil
+}
+
+// makePerPartyAccountsAndTransfers create a party specific per market accounts for bond, margin and fee.
+// It also transfers required commitment amount to per market bond account.
+func (m *MarketLiquidity) makePerPartyAccountsAndTransfers(ctx context.Context, party string, commitmentAmount *num.Uint) error {
+	bondAcc, err := m.collateral.GetOrCreatePartyBondAccount(ctx, party, m.marketID, m.settlementAsset)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.collateral.CreatePartyMarginAccount(ctx, party, m.marketID, m.settlementAsset)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.collateral.GetOrCreatePartyLiquidityFeeAccount(ctx, party, m.marketID, m.settlementAsset)
+	if err != nil {
+		return err
+	}
+
+	// calculates the amount that needs to be moved into the bond account
+	amount, neg := num.UintZero().Delta(commitmentAmount, bondAcc.Balance)
+	ty := types.TransferTypeBondLow
+	if neg {
+		ty = types.TransferTypeBondHigh
+	}
+	transfer := &types.Transfer{
+		Owner: party,
+		Amount: &types.FinancialAmount{
+			Amount: amount.Clone(),
+			Asset:  m.settlementAsset,
+		},
+		Type:      ty,
+		MinAmount: amount.Clone(),
+	}
+
+	tresp, err := m.collateral.BondUpdate(ctx, m.marketID, transfer)
+	if err != nil {
+		m.log.Debug("bond update error", logging.Error(err))
+		return err
+	}
+	m.broker.Send(events.NewLedgerMovements(ctx, []*types.LedgerMovement{tresp}))
+
+	return nil
+}
+
+// AmendLiquidityProvision forwards a LiquidityProvisionAmendment to the Liquidity Engine.
+func (m *MarketLiquidity) AmendLiquidityProvision(
+	ctx context.Context,
+	lpa *types.LiquidityProvisionAmendment,
+	party string,
+	deterministicID string,
+	marketState types.MarketState,
+) error {
+	m.idGen = idgeneration.New(deterministicID)
+	defer func() { m.idGen = nil }()
+
+	if !m.canSubmitCommitment(marketState) {
+		return ErrCommitmentSubmissionNotAllowed
+	}
+
+	if err := m.liquidityEngine.ValidateLiquidityProvisionAmendment(lpa); err != nil {
```

Comment:
this isn't implemented by v2 is it?

Reply:
The function is implemented in V2 engine. Or what do you mean?